adoc/deployment-azure.adoc

```diff
@@ -0,0 +1,204 @@
+
+The AZURE deployment created by our {tf} template files leads to the
+creation of the infrastructure described in the next paragraphs.
+
+==== Network
+
+All of the infrastructure is created inside of a user specified AZURE region.
+The resources are currently all located inside of the same availability
+zone. All the nodes are placed inside of the same virtual network, within the same
+subnet.
+
+Worker nodes are never exposed to the public internet. On the opposite
+each master nodes has a public IP address by default. This allows users to
+connect to them via ssh from their computers.
+
+It's also possible disable this behaviour and make **all** the nodes private.
+This can be done setting the `create_bastionhost` variable to `true`.
+
+When this variable is set all the master nodes cease to have a public IP address.
+An [Azure Bastion](https://docs.microsoft.com/en-us/azure/bastion/bastion-overview)
+instance is created which becomes the only way to ssh into the cluster.
+
+==== Load Balancer
+
+The {tf} template files take care of creating a
+link:https://azure.amazon.com/elasticloadbalancing/[Classic Load Balancer]
+which exposes the Kubernetes API service deployed on the control plane
+nodes.
+
+The load balancer exposes the following ports:
+
+* `6443`: Kubernetes API server
+* `32000`: Dex (OIDC Connect)
+* `32001`: Gangway (RBAC Authenticate)
+
+==== Accessing the nodes
+
+A default `sles` user is created on each node of the cluster. The user has
+administrator capabilities by using the `sudo` utility.
+
+By default password based authentication is disabled. It's possible to log
+using the ssh key specified via the `admin_ssh_key` variable.
+It's also possible to enable password based authentication by specifying a
+value for the `admin_password` variable. Note well: Azure has some security
+checks in place to avoid the usage of weak passwords.
+
+When the bastion host creation is disabled the access to the master nodes of
+the cluster is just a matter of doing a ssh against their public IP address.
+
+Accessing a cluster through an Azure Bastion requires a different procedure.
+
+==== Using Azure Bastion
+
+Azure bastion supports only RSA keys with PEM format. These can be created by
+doing:
+
+```
+ssh-keygen -f azure -t rsa -b 4096 -m pem
+```
+
+This will create a public and private key named `azure`. The **private** key has
+to be provided later to Azure, hence it's strongly recommended to create
+a dedicate key pair.
+
+Once the whole infrastructure is created you can connect into any node of the
+cluster by doing the following steps:
+
+  * Log into Azure portal
+  * Choose one of the nodes of the cluster
+  * Click "connect" and select "bastion" as option
+  * Enter all the required fields
+
+Once this is done a new browser tab will be open with a shell session running
+inside of the desired node.
+
+It's recommended to use Chrome or Chromium during this process. Firefox doesn't
+seem to allow copying and pasting.
```

Comment:
Maybe we can soften the statement about Firefox... hopefully this will get fixed by Azure's team

Maybe we can change that to `As of June 2020, Firefox doesn't ...`

I don't know what @r0ckarong and @nkoranova think about that.

Reply:
Maybe its better to wipe out the statement about Firefox.

Reply:
removed Firefox part